src/util/fork_utils.js

```diff
@@ -73,7 +73,7 @@ function start_workers(metrics_port, count = 0) {
         }
         if (metrics_port > 0) {
             dbg.log0('Starting metrics server', metrics_port);
-            prom_reporting.start_server(metrics_port, true);
+            await prom_reporting.start_server(metrics_port, true);
```

Comment:
@shirady @naveenpaul1 why are we starting it from here and also from the endpoint? 
wouldn't the primary will start it twice?

Reply:
I know,
regardless of that, this prom_reporting.start_server() appears twice
`endpoint.js` - 
```
  // START METRICS SERVER
  if (metrics_port > 0 && cluster.isPrimary) {
      dbg.log0('Starting metrics server', metrics_port);
      await prom_reporting.start_server(metrics_port, false);
      dbg.log0('Started metrics server successfully');
  }
```
and here

Reply:
@romayalon it is:
-  in `fork_utils`: `await prom_reporting.start_server(metrics_port, true);`
- in `endpoint`: `await prom_reporting.start_server(metrics_port, false);`
The variable `fork_enabled` is used in the function and the implementation is different (conditions on this variable).

Reply:
@shirady
correct, 
when forkes_enabled=true -`// Serve all metrics on the root path for system that do have one or more fork running.`
 when forkes_enabled=false `// Serve all metrics on the root path for system that do not have any fork running.`

If I understood the code correctly, we will call it twice, to me it appears to be incorrect, what's the reason you think we need to call if with `await prom_reporting.start_server(metrics_port, false);` when having multiple forks?
@naveenpaul1 can you add your thoughts here as well pls?

Reply:
@romayalon we will start it from endpoint if noobaa running in primary fork 
```
if (metrics_port > 0 && cluster.isPrimary) {
            dbg.log0('Starting metrics server', metrics_port);
            await prom_reporting.start_server(metrics_port, false);
}
```
and fork_util will start it in case of multiple forks

Reply:
thanks @naveenpaul1 I missed the return
@shirady maybe you can add there a comment so it'll be more noticable

Reply:
@naveenpaul1 
I run: `sudo node src/cmd/nsfs --debug 5 --forks 2`
You can see that the primary created `50755` PID and `50756` PID and then each one of them `cluster.isPrimary false`.

> Nov-28 11:35:23.873 [Upgrade/50746]  [WARN] CONSOLE:: WORKER started { id: 1, pid: 50755 }
> Nov-28 11:35:23.874 [Upgrade/50746]  [WARN] CONSOLE:: WORKER started { id: 2, pid: 50756 }
> Nov-28 11:35:23.874 [Upgrade/50746]    [L0] core.util.fork_utils:: Starting metrics server 7004
> Nov-28 11:35:23.875 [Upgrade/50746]    [L0] core.server.analytic_services.prometheus_reporting:: Starting prometheus metrics server on HTTP port 7004

> Nov-28 11:35:26.251 [Upgrade/50755]   [LOG] CONSOLE:: SDSD cluster.isPrimary false
> Nov-28 11:35:26.552 [Upgrade/50756]   [LOG] CONSOLE:: SDSD cluster.isPrimary false

@romayalon, of course I can add a comment - what do you want it to include?

Reply:
that if there is functionality that should be run only from the primary process it should be in the start_forks() function and that we return here.
I wonder if logging/notifications features behave differently on multiple forks

Reply:
I added a comment, I hope it is clearer.
Regarding logging/notifications - we can mention those who worked on it or create a Slack thread about it.